vscode-dotnet-runtime-library/src/Utils/DotnetVersionProvider.ts

```diff
@@ -0,0 +1,65 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import { DotnetVersionSupportStatus, DotnetVersionSupportPhase, IDotnetListVersionsContext, IDotnetListVersionsResult, IDotnetVersion } from '../IDotnetListVersionsContext';
+import { WebRequestWorker } from './WebRequestWorker';
+
+export class DotnetVersionProvider {
+
+    static availableDontetVersionsUrl = 'https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json';
+    static dotnetAvailableVersionsPageUnavailableError = 'The service to request available SDK versions (releases.json) is unavailable.';
+
+    /**
+     * @remarks
+     * Use the release.json manifest (provided to the webWorker) that contains the newest version of the SDK and Runtime for each major.minor of .NET to get the available versions.
+     * Relies on the context listRuntimes to tell if it should get runtime or sdk versions.
+     *
+     * @returns
+     * IDotnetListVersionsResult of versions available.
+     *
+     * @throws
+     * Exception if the API service for releases-index.json is unavailable.
+     */
+    async GetAvailableDotnetVersions(commandContext: IDotnetListVersionsContext | undefined, webWorker: WebRequestWorker)
+    {
+        // If shouldObtainSdkVersions === false, get Runtimes. Else, get Sdks.
+        const shouldObtainSdkVersions : boolean = commandContext?.listRuntimes === null || commandContext?.listRuntimes === undefined || !commandContext.listRuntimes;
+
+        const availableVersions : IDotnetListVersionsResult = [];
+        let response = null;
+
+        try
+        {
+            response = await webWorker.getCachedData();
+        }
+        catch(e)
+        {
+            throw new Error(DotnetVersionProvider.dotnetAvailableVersionsPageUnavailableError);
```

Comment:
I don't remember offhand exactly how telemetry works, but these errors seem like the kind of thing you would want to get a count of in telemetry. Have you checked if they're being logged?

Reply:
They are being logged! Except this error wasn't correctly added to the EventStreams events, nor should it be, since it was duplicated. The telemetry could use a lot of refinement though.